### DISCUSSION TITLE
Potential security issue on the RichEditor in the Filament Form Builder

### DISCUSSION THREAD

**Discussion by mathijso** [filamentphp/filament#13855] [General]
An application i am building has been Pen Tested and in the results it talks about a potential issue with the RichEditor in the Filament Form Builder. I'm not experienced enough to deeply test this myself and solved it now by using ->toolbarButtons([]) and exclude the 'attachFiles' in the list. But maybe this is something that can be researched in more depth.  

The full context:
_"File upload, stored Cross-Site Scripting. The web application is vulnerable to Stored Cross Site Scripting (XSS). The web application contains a component to add attachments to notes via the /livewire/update-file endpoint. The /livewire/update-file endpoint does not perform sufficient checks on the file types of uploads. This makes it possible to add JavaScript code to the relevant product notes via a malicious SVG file.
An attacker, with the ability to create a product note, could use this finding to add a malicious SVG file as an attachment. An attacker can send a link for this SVG file to the victim. The SVG file may contain JavaScript code that is executed when the victim opens the link. This allows an attacker to perform actions on behalf of the victim.
Make sure that user input cannot have any special meaning in the context in which it is used. In the context of file upload functionality, this means that SVG files are not allowed.
The screenshots below show the upload and execution of the Cross- Site Scripting attack. The first screenshot shows an attachment upload for a note page of a product. The red-bordered part shows the URL and the URL path where the upload is generated by the notes page. The blue outlined part shows the content of the file, in our case an SVG with XSS payload. The yellow-bordered part shows that the server accepts it.The second screenshot shows the execution of the Javascript code within the SVG file in the web browser. The red-bordered part shows a pop-up window generated as a result of the Javascript code."_
<img width="805" alt="Screenshot_1" src="https://github.com/user-attachments/assets/d208237f-ef8f-4aaf-8bde-e25ef28ce25d">
<img width="803" alt="Screenshot_2" src="https://github.com/user-attachments/assets/42fe09a4-9241-4484-b623-7f1cecf99177">

Let me know if more context is needed.

**Comment by coufal** [comment]
Some context: https://digi.ninja/blog/svg_xss.php

As far as I understand, a fix would be adding `content-disposition: attachment` to all files in the `storage` directory. This could be done via `.htaccess` or nginx config.

Example

```.htaccess
  <FilesMatch "\.(svg)$">
      Header set Content-Disposition "attachment"
  </FilesMatch>
```

Thanks for bringing it up, I didn't even know that this kind of XSS was possible :D
